### PR TITLE
Fixed nll with label_smoothing to just nll

### LINKED ISSUE #28167: Misleading doc on BLIP `outputs.loss`: doesn't return true NLL but NLL *with label smoothing*
### System Info

Transformers 4.35.2

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Not really a bug, more a misleading feature:
Computing the negative log-likelihood (NLL) is useful for understanding the probability of a caption for a given image, using BLIP generative text decoder. However, if one uses BLIP for ConditionalGeneration as explained here 
https://huggingface.co/docs/transformers/model_doc/blip#transformers.BlipForConditionalGeneration
adapted for computation of the NLL, one would naturally do:
```
from PIL import Image
import requests
from transformers import AutoProcessor, BlipForConditionalGeneration

processor = AutoProcessor.from_pretrained("Salesforce/blip-image-captioning-base")
model = BlipForConditionalGeneration.from_pretrained("Salesforce/blip-image-captioning-base")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)
text = "A image of two cats"

inputs = processor(images=image, text=text, return_tensors="pt")

outputs = model(**inputs, labels=inputs['input_ids'])
nll=outputs.loss.item()
```
However, the loss is computed **with label smoothing** as in training, because it is hard-coded in BLIPLLM head (just like in the original Salesforce code) 

https://github.com/huggingface/transformers/blob/c48787f347bd604f656c2cfff730e029c8f8c1fe/src/transformers/models/blip/modeling_blip_text.py#L892

Therefore it isn't the true NLL that the call to .loss returns, and I believe the documentation should be clearer on this. 
I propose to:
* change the doc to make this clearer 
* or add a parameter label_smoothing when initializing the BLIP model 
* or add a function to compute NLL explicitely, separated from .loss, e.g.:
```
def return_nll(scores, target):
  loss_fct = CrossEntropyLoss(reduction='mean', label_smoothing=0.0) # we're setting it to 0
  loss = loss_fct(scores, target)
  return loss

def compute_generative_probability(model, processor, image, text):

  inputs = processor(images=image, text=text,
                    return_tensors="pt", padding=True)
  outputs = model(**inputs, labels=inputs['input_ids'])
  shifted_predictions_scores = outputs.logits[0 , :-1, :].contiguous()
  shifted_labels = inputs["input_ids"][0, 1:].contiguous().to(shifted_predictions_scores.device)
  nll = return_nll(shifted_predictions_scores, target=shifted_labels)

  return nll
```
Writing this so that others researchers are aware :) Thanks a lot for the amazing library 

### Expected behavior

See code above

### PR DESCRIPTION
**Pull request by nileshkokane01**
# What does this PR do?

This PR fixes #28167 by making label_smoothing= 0 . 

Fixes #28167

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@younesbelkada 

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by younesbelkada — COMMENTED** [review]
Thanks ! Could you apply the same changes on the TF modeling file?

**Comment by nileshkokane01** [comment]
@younesbelkada ,
Can you please review the changes?

File: src/transformers/models/blip/modeling_tf_blip_text.py [hunk 1]
```diff
@@ -1061,7 +1061,7 @@ def call(
             labels = tf.reshape(labels, (-1,))
             # Keras won't give us label smoothing for sparse CE, so we de-sparsify things here
             one_hot_labels = tf.one_hot(labels, depth=self.config.vocab_size, dtype=tf.float32)
-            loss_fct = tf.keras.losses.CategoricalCrossentropy(from_logits=True, label_smoothing=0.1, reduction="none")
+            loss_fct = tf.keras.losses.CategoricalCrossentropy(from_logits=True, label_smoothing=0.0, reduction="none")
```
Comment (younesbelkada) [inline src/transformers/models/blip/modeling_tf_blip_text.py:1064]:
```suggestion
            loss_fct = keras.losses.CategoricalCrossentropy(from_logits=True, label_smoothing=0.0, reduction="none")
```

**Review by younesbelkada — COMMENTED** [review]
Thanks ! there is a small merge conflict with main - could you resolve that ? After that well be able to merge

**Comment by nileshkokane01** [comment]
@younesbelkada  ,
I rebased and resolved the conflict. I hope its a right way, or else let me know.

**Review by younesbelkada — APPROVED** [review]
Thank you @nileshkokane01 for your contribution !

**Review by ArthurZucker — COMMENTED** [review]
IMO that's a breaking change. Not sure if the best is to update the documentation or update this! 
Should we add a config arg for this?

**Comment by ArthurZucker** [comment]
cc @younesbelkada

**Comment by younesbelkada** [comment]
@ArthurZucker IMO this is not really breaking, it is even the opposite as it fixes a bug with respect to training with BLIP- see @NielsRogge ' comment here: https://github.com/huggingface/transformers/issues/28167#issuecomment-1867398976

**Comment by younesbelkada** [comment]
perhaps we can add a warning to users to tell them more about the fix and what they need to do in order to revert to previous behaviour, but I think that it should be alright as it fixes a bug

**Comment by nileshkokane01** [comment]
@younesbelkada sure! I'll do that.

**Comment by younesbelkada** [comment]
thank you @nileshkokane01 !

**Comment by nileshkokane01** [comment]
@younesbelkada ,

Do I have to change the nll loss to the following as well: 

`
loss_fct = CrossEntropyLoss(reduction=reduction, label_smoothing=self.config.lable_smoothing)`

**Comment by younesbelkada** [comment]
hi @nileshkokane01 
Yes please, this sounds great

**Comment by nileshkokane01** [comment]
@younesbelkada ,
can you have a look ?

File: src/transformers/models/blip/configuration_blip.py [hunk 2]
```diff
@@ -94,6 +94,10 @@ class BlipTextConfig(PretrainedConfig):
             Whether the model is used as a decoder.
         use_cache (`bool`, *optional*, defaults to `True`):
             Whether or not the model should return the last key/values attentions (not used by all models).
+        label_smoothing (float, optional): A float in [0.0, 1.0]. Specifies the amount
+            of smoothing when computing the loss, where 0.0 means no smoothing. The targets
+            become a mixture of the original ground truth and a uniform distribution as described in
+            `Rethinking the Inception Architecture for Computer Vision <https://arxiv.org/abs/1512.00567>`__. Default: :math:`0.0`.
```
Comment (younesbelkada) [inline src/transformers/models/blip/configuration_blip.py:100]:
```suggestion
        label_smoothing (float, *optional*):
            A float in [0.0, 1.0]. Specifies the amount of smoothing when computing the loss, where 0.0 means no smoothing. The targets
            become a mixture of the original ground truth and a uniform distribution as described in
            `Rethinking the Inception Architecture for Computer Vision <https://arxiv.org/abs/1512.00567>`__. Default: :math:`0.0`.
```

File: src/transformers/models/blip/configuration_blip.py [hunk 3]
```diff
@@ -298,6 +304,10 @@ class BlipConfig(PretrainedConfig):
             The inital value of the *logit_scale* paramter. Default is used as per the original BLIP implementation.
         image_text_hidden_size (`int`, *optional*, defaults to 256):
             Dimentionality of the hidden state of the image-text fusion layer.
+        label_smoothing (float, optional, *optional*, defaults to 0.0): A float in [0.0, 1.0]. Specifies the amount
+            of smoothing when computing the loss, where 0.0 means no smoothing. The targets
+            become a mixture of the original ground truth and a uniform distribution as described in
+            `Rethinking the Inception Architecture for Computer Vision <https://arxiv.org/abs/1512.00567>`__. Default: :math:`0.0`.
```
Comment (younesbelkada) [inline src/transformers/models/blip/configuration_blip.py:310]:
```suggestion
        label_smoothing (float, optional, *optional*, defaults to 0.0):
            A float in [0.0, 1.0]. Specifies the amount of smoothing when computing the loss, where 0.0 means no smoothing. The targets
            become a mixture of the original ground truth and a uniform distribution as described in
            `Rethinking the Inception Architecture for Computer Vision <https://arxiv.org/abs/1512.00567>`__. Default: :math:`0.0`.
```

**Review by younesbelkada — APPROVED** [review]
Looking good thanks ! left few nits on docstrings!

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing and making this controllable!
